actors/market/tests/market_actor_test.rs

```diff
@@ -235,195 +270,251 @@ fn account_actor_check() {
+    withdraw_client_balance(&mut rt, withdraw_amount, amount, client_addr);
 
-    assert_eq!(get_escrow_balance(&rt, &client_addr).unwrap(), TokenAmount::from(19u8));
+    assert_eq!(get_escrow_balance(&rt, &client_addr).unwrap(), TokenAmount::from(0));
 }
 
-#[ignore]
 #[test]
-fn client_withdraw_more_than_available() {
+fn worker_withdrawing_more_than_escrow_balance_limits_to_available_funds() {
     let mut rt = setup();
 
-    let client_addr = Address::new_id(CLIENT_ID);
-
-    let amount = TokenAmount::from(20u8);
-    add_participant_funds(&mut rt, client_addr, amount.clone());
+    let provider_addr = Address::new_id(PROVIDER_ID);
+    let owner_addr = Address::new_id(OWNER_ID);
+    let worker_addr = Address::new_id(WORKER_ID);
 
-    rt.set_caller(*ACCOUNT_ACTOR_CODE_ID, client_addr);
-    rt.expect_validate_caller_type(vec![*ACCOUNT_ACTOR_CODE_ID, *MULTISIG_ACTOR_CODE_ID]);
+    let amount = TokenAmount::from(20);
+    add_provider_funds(&mut rt, amount.clone(), provider_addr, owner_addr, worker_addr);
 
-    let withdraw_amount = TokenAmount::from(25u8);
+    assert_eq!(get_escrow_balance(&rt, &provider_addr).unwrap(), amount);
 
-    rt.expect_send(
-        client_addr,
-        METHOD_SEND,
-        RawBytes::default(),
+    let withdraw_amount = TokenAmount::from(25);
+    withdraw_provider_balance(
+        &mut rt,
+        withdraw_amount,
         amount,
-        RawBytes::default(),
-        ExitCode::OK,
+        provider_addr,
+        owner_addr,
+        worker_addr,
     );
 
-    let params = WithdrawBalanceParams { provider_or_client: client_addr, amount: withdraw_amount };
-
-    assert!(rt
-        .call::<MarketActor>(Method::WithdrawBalance as u64, &RawBytes::serialize(params).unwrap(),)
-        .is_ok());
-
-    rt.verify();
-
-    assert_eq!(get_escrow_balance(&rt, &client_addr).unwrap(), TokenAmount::from(0u8));
+    assert_eq!(get_escrow_balance(&rt, &provider_addr).unwrap(), TokenAmount::from(0));
+    // TODO: actor.checkState(rt)
 }
 
-#[ignore]
 #[test]
-fn worker_withdraw_more_than_available() {
+fn deal_starts_on_day_boundary() {
+    let start_epoch = DEAL_UPDATES_INTERVAL; // 2880
+    let end_epoch = start_epoch + 200 * EPOCHS_IN_DAY;
+    let publish_epoch = ChainEpoch::from(1);
+
     let mut rt = setup();
+    rt.set_epoch(publish_epoch);
 
+    let client_addr = Address::new_id(CLIENT_ID);
+    let provider_addr = Address::new_id(PROVIDER_ID);
     let owner_addr = Address::new_id(OWNER_ID);
     let worker_addr = Address::new_id(WORKER_ID);
-    let provider_addr = Address::new_id(PROVIDER_ID);
-
-    let amount = TokenAmount::from(20u8);
-    add_provider_funds(&mut rt, provider_addr, owner_addr, worker_addr, amount.clone());
 
-    assert_eq!(amount, get_escrow_balance(&rt, &provider_addr).unwrap());
+    for i in 0..(3 * DEAL_UPDATES_INTERVAL) {
+        let piece_cid = make_piece_cid((format!("{i}")).as_bytes());
+        //println!("{i}: {}", piece_cid);
```

Comment:
This kind of commented prints are really falling in the category "code hoarding" and must be removed (since Code Hoarding is a deep and overwhelming mess).

And I would say there is little interest to keep prints or debug logs in unit tests.